src/ctia/entity/relationship.clj

```diff
@@ -71,51 +72,75 @@
   (st/merge PagingParams
             RelationshipFieldsParam))
 
-(s/defschema IncidentCasebookLinkRequest
-  {:casebook_id Reference
+(s/defschema IncidentLinkRequest
+  {;; Exactly one of:
+   (s/optional-key :casebook_id) Reference
+   (s/optional-key :investigation_id) Reference
+
+   ;; Extra keys
    (s/optional-key :tlp) TLP})
 
-(def incident-casebook-link-route
+(def incident-link-route
   (POST "/:id/link" []
         :return rs/Relationship
-        :body [link-req IncidentCasebookLinkRequest
+        :body [link-req IncidentLinkRequest
                {:description "an Incident Link request"}]
-        :summary "Link an Incident to a Casebook"
+        :summary "Link an Incident to a Casebook or Investigation"
         :path-params [id :- s/Str]
+        ;; :read-<source-type> is checked while processing the body
         :capabilities #{:read-incident
-                        :read-casebook
                         :read-relationship
                         :create-relationship}
         :auth-identity identity
         :identity-map identity-map
-        (let [incident (read-store :incident
+        (let [source-type-kw (let [types (select-keys link-req
+                                                      [:casebook_id
+                                                       :investigation_id])]
+                               (when-not (= 1 (count types))
+                                 (bad-request! {:error "Missing source type"}))
```

Comment:
```suggestion
                                 (bad-request! {:error "Please provide a casebook_id OR an investigation_id"}))
```

The current error message is not very explicit.

Reply:
Could you test the cases where:
1. `casebook_id` and `investigation_id` are provided
2. none of them are provided